### PR TITLE
dev/core#5343 Fix failure to load membership custom data on initial load

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#5343 Fix failure to load membership custom data on initial load

Before
----------------------------------------
As described in https://lab.civicrm.org/dev/core/-/issues/5343

After
----------------------------------------
Loads on initial load

Technical Details
----------------------------------------
The membership type was not assigned

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Member/Form/Membership.php [hunk 1]
```diff
@@ -224,7 +224,7 @@ public function preProcess() {
     }
 
     $this->assign('customDataType', 'Membership');
-    $this->assign('customDataSubType', $this->_memType);
+    $this->assign('customDataSubType', $this->getMembershipValue('membership_type_id'));
```
Comment (seamuslee001) [inline CRM/Member/Form/Membership.php:227]: @eileenmcnaughton would this work when creatng a membership?

Comment (eileenmcnaughton) [reply CRM/Member/Form/Membership.php:227]: @seamuslee001 I tested in create mode & it correctly returns NULL from this function

Comment (justinmosier) [reply CRM/Member/Form/Membership.php:227]: This fix works for me on a dev copy running Civi 5.75.0 on Drupal 10.3.1.

Comment (seamuslee001) [reply CRM/Member/Form/Membership.php:227]: ok guess this is fine @eileenmcnaughton I was just noticing a few lines up that memType was getting set in some cases still but i'm guessing the JS will handle that so ok to merge

**Pull request merged**